tools/perfscale-audit/metric-client/metric-client.go

```diff
@@ -0,0 +1,169 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package metricclient
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"log"
+	"math"
+	"net/http"
+	"time"
+
+	audit_api "kubevirt.io/kubevirt/tools/perfscale-audit/api"
+
+	api "github.com/prometheus/client_golang/api"
+	apiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+)
+
+type transport struct {
+	transport http.RoundTripper
+	userName  string
+	password  string
+	token     string
+}
+
+func (a transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if a.userName != "" {
+		req.SetBasicAuth(a.userName, a.password)
+	}
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", a.token))
+	return a.transport.RoundTrip(req)
+}
+
+type MetricClient struct {
+	client apiv1.API
+	cfg    *audit_api.InputConfig
+}
+
+func NewMetricClient(cfg *audit_api.InputConfig) (*MetricClient, error) {
+
+	url := cfg.PrometheusURL
+	token := cfg.PrometheusBearerToken
+	userName := cfg.PrometheusUserName
+	password := cfg.PrometheusPassword
+	tlsVerify := cfg.PrometheusVerifyTLS
+
+	apiCfg := api.Config{
+		Address: url,
+		RoundTripper: transport{
+			transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: tlsVerify}},
+			token:     token,
+			userName:  userName,
+			password:  password,
+		},
+	}
+	c, err := api.NewClient(apiCfg)
+	if err != nil {
+		return nil, err
+	}
+	api := apiv1.NewAPI(c)
+
+	_, err = api.Config(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+	log.Print("Established connection with prometheus endpoint.")
+
+	return &MetricClient{client: api, cfg: cfg}, nil
+}
+
+func (m *MetricClient) query(query string) (model.Value, error) {
+	log.Printf("Makeing query [%s]", query)
+	val, _, err := m.client.Query(context.TODO(), query, *m.cfg.EndTime)
```

Comment:
I think it might be better to use Prometheus QueryRange, as [here](https://github.com/prometheus/client_golang/blob/master/api/prometheus/v1/example_test.go#L70).
With QueryRange, we can define the time interval to collect the metrics. This will enable us to run the tool to collect the results of a specific experiment, or exclude experiments from the analysis.

Reply:
> This will enable us to run the tool to collect the results of a specific experiment

This is what the Query does when i give it a timestamp along with a rate with a duration